packages/flutter_tools/lib/src/ios/device_ios.dart

```diff
@@ -546,52 +546,48 @@ Future<bool> _buildIOSXcodeProject(ApplicationPackage app, bool isDevice) async
   }
 }
 
-bool enabled = false;
 Future _addServicesToBundle(Directory bundle) async {
-  if (enabled) {
-    List<Map<String, String>> services = [];
-    await parseServiceConfigs(services);
-    await _fetchFrameworks(services);
-    _copyFrameworksToBundle(bundle.path, services);
-
-    generateServiceDefinitions(bundle.path, services, ios: true);
-  }
+  List<Map<String, String>> services = [];
+  printTrace("Trying to resolve native pub services");
+
+  // Step 1: Parse the service configuration yaml files present in the service
+  //         pub packages.
+  await parseServiceConfigs(services);
+  printTrace("Found ${services.length} service definition(s)");
+
+  // Step 2: Copy framework dylibs to the correct spot for xcodebuild to pick up.
+  Directory frameworksDirectory = new Directory(path.join(bundle.path, "Frameworks"));
+  _copyServiceFrameworks(services, frameworksDirectory);
+
+  // Step 3: Copy the service definitions manifest at the correct spot for
+  //         xcodebuild to pick up.
+  File manifestFile = new File(path.join(bundle.path, "ServiceDefinitions.json"));
+  _copyServiceDefinitionsManifest(services, manifestFile);
 }
 
-Future _fetchFrameworks(List<Map<String, String>> services) async {
+void _copyServiceFrameworks(List<Map<String, String>> services, Directory frameworksDirectory) {
+  printTrace("Copying service frameworks to '${path.absolute(frameworksDirectory.path)}'");
+  frameworksDirectory.createSync(recursive: true);
   for (Map<String, String> service in services) {
-    String frameworkUrl = service['framework'];
-    service['framework-path'] = await getServiceFromUrl(
-       frameworkUrl, service['root'], service['name'], unzip: true);
-  }
-}
-
-void _copyFrameworksToBundle(String destDir, List<Map<String, String>> services) {
-  // TODO(mpcomplete): check timestamps.
-  for (Map<String, String> service in services) {
-    String basename = path.basename(service['framework-path']);
-    String destPath = path.join(destDir, basename);
-    _copyDirRecursive(service['framework-path'], destPath);
+    File dylib = new File(path.join(service['root'], service['ios-framework']));
```

Comment:
Don't do this. Use the getServiceFromUrl helper from services.json. It knows how to fetch a remote URL and cache it locally. (It returns the path to the local file).